### PR TITLE
Allow resubmission of blocks with the same signing root

### PR DESCRIPTION
**Pull request by shayzluf**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**
- allow resigning of block with same signing root by not detecting it as slashble
- in case of zerohash signing root any attestation should be slashable

**Which issues(s) does this PR fix?**

Part of #7813

**Other notes for review**

### CODE REVIEW

File: validator/slashing-protection/local/proposer_protect.go [hunk 1]
```diff
@@ -25,7 +25,9 @@ func (s *Service) IsSlashableBlock(
 		return false, errors.Wrap(err, "failed to get proposal history")
 	}
 	// Check if we are performing a double block proposal.
-	if existingSigningRoot != nil && !bytes.Equal(existingSigningRoot, params.BeaconConfig().ZeroHash[:]) {
+	same := existingSigningRoot != nil && bytes.Equal(existingSigningRoot, signingRoot[:])
+	if existingSigningRoot != nil && !same {
+		fmt.Printf("existingSigningRoot %#x signingRoot %#x\n", existingSigningRoot[26:], signingRoot[26:])
```
Comment (prestonvanloon) [inline validator/slashing-protection/local/proposer_protect.go:30]: ??

Comment (shayzluf) [reply validator/slashing-protection/local/proposer_protect.go:30]: left over from debugging. will look through again

Comment (shayzluf) [reply validator/slashing-protection/local/proposer_protect.go]: removed

File: validator/slashing-protection/local/proposer_protect.go [hunk 2]
```diff
@@ -25,7 +24,8 @@ func (s *Service) IsSlashableBlock(
 		return false, errors.Wrap(err, "failed to get proposal history")
 	}
 	// Check if we are performing a double block proposal.
-	if existingSigningRoot != nil && !bytes.Equal(existingSigningRoot, params.BeaconConfig().ZeroHash[:]) {
+	same := existingSigningRoot != nil && bytes.Equal(existingSigningRoot, signingRoot[:])
+	if existingSigningRoot != nil && !same {
```
Comment (rauljordan) [inline validator/slashing-protection/local/proposer_protect.go:28]:
```suggestion
	hasSameSigningRoot := existingSigningRoot != nil && bytes.Equal(existingSigningRoot, signingRoot[:])
	if !hasSameSigningRoot {
```

**Comment by rauljordan** [comment]
Closing in favor of #8085

**Pull request closed without merging**